aloha-core/src/main/scala/com/eharmony/aloha/models/exploration/BootstrapModel.scala

```diff
@@ -0,0 +1,108 @@
+package com.eharmony.aloha.models.exploration
+
+import com.eharmony.aloha.factory.{ModelFactory, ModelParser, ParserProviderCompanion}
+import com.eharmony.aloha.id.ModelIdentity
+import com.eharmony.aloha.models.{Model, BaseModel}
+import com.eharmony.aloha.score.Scores.Score
+import com.eharmony.aloha.score.basic.ModelOutput
+import com.eharmony.aloha.score.conversions.ScoreConverter
+import com.eharmony.aloha.score.conversions.ScoreConverter.Implicits.IntScoreConverter
+import com.eharmony.aloha.semantics.Semantics
+import com.eharmony.aloha.semantics.func.GenAggFunc
+import com.mwt.explorers.BootstrapExplorer
+import com.mwt.policies.Policy
+
+import scala.collection.JavaConversions.seqAsJavaList
+import scala.collection.{immutable => sci}
+
+case class NumberedPolicy(index: Int) extends Policy[sci.IndexedSeq[Int]] {
+  override def chooseAction(actions: sci.IndexedSeq[Int]): Int = actions(index)
+}
+
+/**
+  * Created by jmorra on 2/26/16.
+  */
+case class BootstrapModel[A, B](
+  modelId: ModelIdentity,
+  models: sci.IndexedSeq[Model[A, Int]],
+  salt: GenAggFunc[A, Long],
+  classLabels: sci.IndexedSeq[B])(implicit scB: ScoreConverter[B]) extends BaseModel[A, B] {
+
+  @transient lazy val explorer = new BootstrapExplorer[sci.IndexedSeq[Int]](
+    models.indices.map(i => NumberedPolicy(i): Policy[sci.IndexedSeq[Int]]),
+    classLabels.size
+  )
+
+  /** Produce a score.
+    * @param a an input to the model representing covariate data.
+    * @param audit Whether the second field of the result Tuple2 should be Some (true) or None (false)
+    * @return a Tuple2 whose first field represents a simple version of the score, the second field (that should be
+    *         a Some instance if audit is true) is a more involved reporting of the score including errors and all
+    *         sub-model scores.
+    */
+  override private[aloha] def getScore(a: A)(implicit audit: Boolean): (ModelOutput[B], Option[Score]) = {
+    // The basic idea here is that we want a failure or a list of successes, so the Either comes first and the results
+    // are all on the right.  If there is any failure then the Left will be evaluated, else all the successes will be
+    // combined into one IndexedSeq on the right.
+    val mos = models.foldLeft(Right(sci.IndexedSeq()):
+      Either[(ModelOutput[B], Option[Score]), sci.IndexedSeq[(Int, Option[Score])]]) { (x, y) =>
```

Comment:
Seeing how this is used on lines 58, 61, I would make the right side a `Right [(sci.IndexedSeq[Int],  Option[sci.IndexedSeq[Score])]`.  Then you don't need to `map` / `flatMap` later.